bool flash
float32 ratio_p_t
float32 ratio_delta
float32 volume
float32 volume_delta
